tests/generics/tmacroinjectedsym.nim

```diff
@@ -113,3 +113,62 @@ block: # issue #22605, original complex example
     "ok"
 
   doAssert g2(int) == "error"
+
+block: # issue #23865
+  type Xxx = enum
+    error
+    value
+
+  type
+    Result[T, E] = object
```

Comment:
lol we really need to stop copy-pasting this version of `Result` that has all the previous nim bug workarounds in it ;)

Reply:
Making the test cases reflect real world code has its advantages too.